encoding.go

```diff
@@ -140,35 +143,44 @@ func parseSkipList(serialized []byte, depth byte) (VerkleNode, error) {
 	var values [NodeWidth][]byte
 	offset := leafStemOffset + StemSize + 3*banderwagon.UncompressedSize // offset in the serialized payload
 	valueIdx := 0                                                        // Index of the value being deserialized
-	for valueIdx < NodeWidth {
-		rangecount := serialized[offset+1]
-		gapsize := serialized[offset]
-		valueIdx += int(gapsize)
-		offset += 2
-		for i := 0; i < int(rangecount); i++ {
-			values[valueIdx] = serialized[offset : offset+leafSlotSize]
+
+	// shortcut: the leaf is full and so both values are 0.
+	if serialized[offset] == 0 && serialized[offset+1] == 0 {
```

Comment:
Is this worth it? If the node is full, doesn't sound we should serialize with skip list really. Sounds wasteful.